plugins/org.polarsys.capella.scenario.editor.embeddededitor/src/org/polarsys/capella/scenario/editor/embeddededitor/commands/XtextToDiagramCommands.java

```diff
@@ -445,10 +459,92 @@ private static void editElements(Scenario scenario, EList<EObject> elements) {
             editElements(scenario, operand.getBlock().getBlockElements());
           }
         }
+      } else if (xtextElement instanceof org.polarsys.capella.scenario.editor.dsl.textualScenario.StateFragment) {
+
+        editStateFragment(scenario,
+            (org.polarsys.capella.scenario.editor.dsl.textualScenario.StateFragment) xtextElement);
       }
     }
   }
   
+  private static void editStateFragment(Scenario scenario,
+      org.polarsys.capella.scenario.editor.dsl.textualScenario.StateFragment xtextElement) {
+
+    String timeline = ((org.polarsys.capella.scenario.editor.dsl.textualScenario.StateFragment) xtextElement)
+        .getTimeline();
+    String name = ((org.polarsys.capella.scenario.editor.dsl.textualScenario.StateFragment) xtextElement).getName();
+    String keyword = ((org.polarsys.capella.scenario.editor.dsl.textualScenario.StateFragment) xtextElement)
+        .getKeyword();
+
+    InstanceRole instanceRole = EmbeddedEditorInstanceHelper.getInstanceRole(timeline);
+
+    EList<TimeLapse> ownedTimeLapses = scenario.getOwnedTimeLapses();
+
+    switch (keyword) {
+    case DslConstants.STATE:
+      AbstractState abstractState = getCorrespondingCapellaAbstractState(scenario, instanceRole, name);
+      List<TimeLapse> ownedStateFragments = ownedTimeLapses.stream()
```

Comment:
I will merge this commit, but when adding mode, function support remove all the checks outside getCorrespondingCapellaAbstractState. Pass the keyword as parameter to getCorrespondingCapellaAbstractState and in filter condition, check there that the keyword is matched also.